File: android/app/src/main/java/com/amia_mobile/ReactModules/AudioModule/AudioModule.java
package com.amia_mobile;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import java.util.Map;
import java.util.HashMap;
import android.widget.Toast;
import android.app.AlertDialog;
import com.facebook.react.bridge.UiThreadUtil;

public class AudioModule extends ReactContextBaseJavaModule {
    AudioModule(ReactApplicationContext context) {
        super(context);
    }

    @Override
    public String getName() {
        return "AudioModule";
    }

    @ReactMethod
    public void logAudio(String name, String location) {
        final String message = "Create event called with name: " + name + " and location: " + location;

        UiThreadUtil.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                new AlertDialog.Builder(getCurrentActivity())
                        .setTitle("Event Info")
                        .setMessage(message)
                        .setPositiveButton(android.R.string.ok, null)
                        .show();
            }
        });
    }

}

File: android/app/src/main/java/com/amia_mobile/ReactModules/AudioModule/AudioModule.kt
package com.amia_mobile

import android.media.MediaRecorder
import com.facebook.react.bridge.*
import java.io.IOException

class AudioModule(private val reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    private var recorder: MediaRecorder? = null
    private var currentFilePath: String? = null

    override fun getName(): String {
        return "AudioModule"
    }

    @ReactMethod
    fun startRecording(promise: Promise) {
        currentFilePath = "${reactContext.filesDir.absolutePath}/audioRecording_${System.currentTimeMillis()}.mp4"

        recorder = MediaRecorder().apply {
            setAudioSource(MediaRecorder.AudioSource.MIC)
            setOutputFormat(MediaRecorder.OutputFormat.MPEG_4)
            setAudioEncoder(MediaRecorder.AudioEncoder.AMR_WB)
            setOutputFile(currentFilePath)
            try {
                prepare()
                start()
                promise.resolve(null)
            } catch (e: IOException) {
                promise.reject("START_RECORDING_FAILED", e)
            }
        }
    }

    @ReactMethod
    fun stopRecording(promise: Promise) {
        try {
            recorder?.apply {
                stop()
                reset()
                release()
            }
            recorder = null
            promise.resolve(currentFilePath)
        } catch (e: RuntimeException) {
            promise.reject("STOP_RECORDING_FAILED", e)
        }
    }

    @ReactMethod
    fun pauseRecording(promise: Promise) {
        try {
            recorder?.pause()
            promise.resolve(null)
        } catch (e: RuntimeException) {
            promise.reject("PAUSE_RECORDING_FAILED", e)
        }
    }

    @ReactMethod
    fun resumeRecording(promise: Promise) {
        try {
            recorder?.resume()
            promise.resolve(null)
        } catch (e: RuntimeException) {
            promise.reject("RESUME_RECORDING_FAILED", e)
        }
    }
}
